base/mmap.jl

```diff
@@ -57,9 +57,9 @@ grow!(::Anonymous,o::Integer,l::Integer) = return
 function grow!(io::IO, offset::Integer, len::Integer)
     pos = position(io)
     filelen = filesize(io)
-    if filelen < offset + len
```

Comment:
this also changes the behavior so that `grow!` also truncates the file if it is longer than `len`, however, it is not clear to me that change is necessary here (other than that it might occur anyways from the race condition between calling filesize and call ftruncate).